### DISCUSSION TITLE
TShock 5.2 for Terraria 1.4.4.9

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2942] [Announcements]
Welcome to TShock for Terraria 5.2 with support for Terraria 1.4.4.9. This is, of course, also on nuget, but it may take up to an hour to appear.

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

This release was requested by @drunderscore! Thank you so, so much for reaching out to get an update.

## Notable changes in this release
* An additional option `pvpwithnoteam` is added at `PvPMode` to enable PVP with no team. (@CelestialAnarchy, #2617, @ATFGK)
* Corrected and updated deserialization of the following packets (@ATFGK):
  * `ProjectileNew`: Read the third `AI` value.
    * Before this change, it was previously possible for the projectile damage limit to falsely trigger, such as when using the Terra Balde and Fire Gauntlet together.
  * `PlayerSpawn`: Read the `NumberOfDeathsPVE` and `NumberOfDeathsPVP` values.
    * Before this change, the `PlayerSpawnContext` was always read incorrectly, due to the values above being placed in the middle of the existing structure.
  * `NpcTeleportPortal`: Read the NPC index as a `ushort` instead of a `byte`.
  * `PlaceObject`: Read the `Random` value.
    * Before this change, the `Direction` was always read incorrectly, due to the value above being placed in the middle of the existing structure.
  * `Zones`: Read the `zone5` value.
  * `PaintTile` and `PaintWall`: Read the `coatTile` and `coatWall` values.
  * `PlayerHurtV2`: Read the `cooldownCounter` value.
* Updated `SpawnMsg` to include the `NumberOfDeathsPVE` and `NumberOfDeathsPVP`, and allow them to be optionally used in `TSPlayer.Spawn`. (@ATFGK)
* Added `WorldTileProvider` to the tshock config with values `default`, `constileation` or `heaptile`. This allows tile providers to be changed in environments where CLI args cannot be altered. See the documentation website for more info about these providers. (@SignatureBeef)
* Updated the Utils.FindByIdOrName to follow same logic. Now fuzzy match fallback to `StartsWith` and then `Contains`. (@sgkoishi)
* Added `ShadowCandle` and `BrainOfConfusionBuff` (BoC dodge buff) to the `PlayerAddBuffWhitelist` (@drunderscore)
* Improved rejection message and code duplication in `OnPlayerBuff`. (@drunderscore)
  * This will make it so Bouncer rejections regarding `PlayerAddBuff` will now always include the sender index, buff type, receiver index, and time in ticks, allowing much faster triage of buff whitelist issues.
* Allowed Digging Molecart and bomb fish to break tiles and place tracks. (@sgkoishi)
* Added built-in package management capabilities for plugins. (@pontaoski)
* Fixed Super Sponge unable to absorb shimmer. (@sgkoishi, #2833)
* Increased whitelisted duration of the Mighty Wind (`WindPushed`) buff (from sandstorms). (@drunderscore)
* Allowed the Hellfire (`OnFire3`) buff. (@drunderscore)
* Allowed Digging Molecart and bomb fish to break tiles and place tracks (@sgkoishi)
* Initialized achievements and the `AchievementManager` on the server. This ensures that players cannot cause exceptions to be thrown, chat messages are always logged, and allows achievement names to be localized in the console. Also added a test case for this. (@drunderscore)
* Allowed multiple test cases to be in TShock's test suite. (@drunderscore)
* Fixed unable to use Purification/Evil Powder in jungle. (@sgkoishi)
* Set the `GetDataHandledEventArgs.Player` property for the `SyncTilePicking` data handler. (@drunderscore)
* Relaxed custom death message restrictions to allow Inferno potions in PvP. (@drunderscore)
* Allowed Flower Boots to place Ash Flowers on Ash Grass blocks. (@punchready)
* Removed unnecessary range check that artifically shortened quick stack reach. (@boddyn, #2885, @bcat)
* Improved the exploit protection in tile rect handling. (@punchready)
* Additional translation updates via Crowdin contributors from the past several months.
## Upgrading

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

To upgrade TShock, replace all of the files in the location of your server installation with all of the files from the release zip we provide. Overwrite all files. No data will be lost.

Special note: the arm (x86) version is not provided. Due to serious issues with these versions that make TShock work sub-optimally, we don't suggest them anymore.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/'>TShock 5.2 for Terraria 1.4.4.9</a>.</em>

**Comment by thank-u-tshock** [comment]
i made an account just to say thank you for this, i can finally smoothly from in house, host a server to play with my cousin from home in iraq. thank you for this

**Comment by MaksDragon** [comment]
good work guys!!

**Comment by hieu200219** [comment]
Pls help me why i don’t see world.wld and how to restore world on linux 20.04

**Comment by 123mjb** [comment]
When I use the command **./TShock.Server** the error 
**bash: ./TShock.Server: cannot execute: required file not found**
is returned.
I am using a raspberry Pi 5 and the arm 64 version. What am i missing?

**Comment by KotofeyBarsikovich** [comment]
Please make a x86 bit version

**Reply by sgkoishi** [reply]
TShock by itself is platform-independent not the bottleneck, but some of the dependencies are not working well with 32-bit. Most CPUs within the recent 15 years are x64 - are you using devices from decades ago?

**Comment by Dorito21King** [comment]
uhh
when i use the tshock installer it shows me this
![image_2024-11-16_142645448](https://github.com/user-attachments/assets/351402a4-2adf-4cdf-9b9f-edc7b5f322bd)
can someone help me?

**Reply by sgkoishi** [reply]
Perhaps you didn't extract all the files in the zip. You need all of them, not just the installer.

**Reply by Dorito21King** [reply]
well i did that 
![image_2024-11-16_143026902](https://github.com/user-attachments/assets/99720612-7539-44a1-91fc-c87649e502ff)
and it says the same message anyways😢

**Reply by sgkoishi** [reply]
I don't think you get the installer from this zip. Extract all the files in the zip which you get your installer from.

**Reply by Dorito21King** [reply]
this one?
![image_2024-11-16_143625891](https://github.com/user-attachments/assets/c77664db-6c3c-498c-8c2f-c78bf1e642eb)
it only has some sort of CS files i cant open

**Reply by sgkoishi** [reply]
Is there a TShock.Installer.exe inside?

**Reply by Dorito21King** [reply]
nope only cs files

**Reply by Dorito21King** [reply]
even in the folder called TShockInstaller it only has cs files

**Reply by sgkoishi** [reply]
As I said earlier, find the zip which you get your installer from.

**Reply by Dorito21King** [reply]
the only one that i see a tshock installer is that one that got the error other than that i have none

**Reply by sgkoishi** [reply]
Yes. Is it the only one file in that zip?

**Reply by Dorito21King** [reply]
the only file named tshock installer in the folders

**Reply by Dorito21King** [reply]
this folder here >  ![image_2024-11-16_143026902](https://private-user-images.githubusercontent.com/188706921/386869728-99720612-7539-44a1-91fc-c87649e502ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3ODQ3MDgsIm5iZiI6MTczMTc4NDQwOCwicGF0aCI6Ii8xODg3MDY5MjEvMzg2ODY5NzI4LTk5NzIwNjEyLTc1MzktNDRhMS05MWZjLWM4NzY0OWU1MDJmZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNlQxOTEzMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMzY0MTA3ZDMwNTdiM2Q3MWJlN2Q3NjM4YzA5OTExMDhhOTc5YmE4NDg2OGRlYmI0YTQ3YWU1MGFiNTYxODZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qUt2j_ZyakBoakYqAUUBc4umznsfLuXDUxEQ2s7HyxE)

**Reply by sgkoishi** [reply]
Do you mean there is only one FILE in the zip, and that file is called TShock.Installer.exe? If so, you are downloading the wrong stuff. None of the official release of TShock looks like this.

**Reply by Dorito21King** [reply]
uhh i just downloaded this one https://github.com/Pryaxis/TShock/releases
 @sgkoishi

**Reply by sgkoishi** [reply]
Is it a zip file containing `TShock.Installer`?

**Reply by Dorito21King** [reply]
yes

**Reply by sgkoishi** [reply]
As I said in the second comment: Extract all the files in **the zip which you get your installer from**. Did you get your installer from this file?

**Reply by Dorito21King** [reply]
yes but it wont work

**Reply by sgkoishi** [reply]
You've never done this, because you didn't extract `TShock-5.2-for-Terraria-1.4.4.9-win-x64-Release.zip`, but only extract the `dotnet-runtime-6.0.11-win-x64` which nobody told you to do so.

**Reply by Dorito21King** [reply]
i did both? they both are extracted but if the dotnet one is the one causing problems i can delete it and leave only the zipped one

**Reply by sgkoishi** [reply]
If you extract `TShock-5.2-for-Terraria-1.4.4.9-win-x64-Release.zip` properly, you should see `bin`, `i18n`, `ServerPlugins`, `GeoIP.dat`, `TShock.Installer.exe` and `TShock.Server.exe`.

From none of the screenshot shows you did extract it. You said you have one with lots of CS file and the dotnet one - they are not `TShock-5.2-for-Terraria-1.4.4.9-win-x64-Release.zip`.

**Reply by Dorito21King** [reply]
![image_2024-11-16_153621734](https://github.com/user-attachments/assets/5cee38d3-7672-4b1b-8b7a-8d8f9ba94014)
i did do that and it stays the same thing as i showed u the screenshot before only dotnet,dotnet-runtime-6.0.11-win-x64 and the buggy installer

**Reply by sgkoishi** [reply]
Do you mind record a video?

**Reply by Dorito21King** [reply]
im heading to a restaurant now i can do it later though.

**Reply by sgkoishi** [reply]
Sure. When you are available, please do show that you have `bin`, `i18n`, `ServerPlugins`, `GeoIP.dat`, `TShock.Installer.exe` and `TShock.Server.exe` these six items in the _current_ folder (it doesn't matters if you have other stuff like dotnet), and double click the `TShock.Installer.exe`.
